core/src/miner/mem_pool.rs

```diff
@@ -792,8 +831,47 @@ impl MemPool {
         }
 
         // We might have filled a gap - move some more parcels from future
-        self.move_matching_future_to_current(signer_public, seq, state_seq);
-        self.move_matching_future_to_current(signer_public, seq + U256::one(), state_seq);
+        self.move_matching_future_to_current(signer_public, seq, state_seq, &parcel.insertion_time, timestamp);
+        self.move_matching_future_to_current(
+            signer_public,
+            seq + U256::one(),
+            state_seq,
+            &parcel.insertion_time,
+            timestamp,
+        );
+
+        if Self::should_wait_timelock(&parcel.timelock, parcel.insertion_time, timestamp) {
+            // Check same seq is in current. If it
+            // is than move the following current items to future.
+            let mut moved_to_future_flag = false;
+            let best_block_number = parcel.insertion_time;
+            if let Some(_) = self.current.by_signer_public.get(&signer_public, &seq) {
```

Comment:
```rust
            let moved_to_future_flag = if let Some(_) = self.current.by_signer_public.get(&signer_public, &seq) {
                self.move_all_to_future(&signer_public, state_seq);
                true
            } else {
                false
            };
````

Reply:
I edited.

Reply:
Sorry to bother you, but the below code looks better.
```rust
let moved_to_future_flag = self.current.by_signer_public.get(&signer_public, &seq).is_some();
if moved_to_future_flag {
    self.move_all_to_future(&signer_public, state_seq);
}
```